code/game/machinery/bots/secbot.dm

```diff
@@ -326,6 +328,31 @@ Auto Patrol: []"},
 		bot_process() //ensure bot quickly responds
 // look for a criminal in view of the bot
 
+/obj/machinery/bot/secbot/proc/can_boost()
+	var/obj/item/weapon/circuitboard/bot_upgrade_boost/B = locate(/obj/item/weapon/circuitboard/bot_upgrade_boost) in upgrades
+
+	if(B)
+		return !B.boost
+
+/obj/machinery/bot/secbot/proc/activate_boost()
+	var/obj/item/weapon/circuitboard/bot_upgrade_boost/B = locate(/obj/item/weapon/circuitboard/bot_upgrade_boost) in upgrades
+	
+	if(B)
+		B.boost = 1
+		movement_delay = B.boost_delay
+		
+		spawn(B.boost_length * 10)
```

Comment:
would be better to just make the var *10 when setting it,just add a comment to it saying that it's a value in deciseconds